### PR TITLE
Market Analysis: Fix unhandled exceptions and rewrite/fix all unit tests

### PR DESCRIPTION
**Pull request by laxdog**
Fixes the pandas issue of mapping to an axes that isn't the right size and some other unhandled errors when you receive some garbage data.

Completely new set of unit tests. Still missing a few things though. 

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] **I have read CONTRIBUTING.md**
- [X] **I fully understand [Github Flow.](https://guides.github.com/introduction/flow/)**
- [X] **My code adheres to the [code style of this project.](https://poloniexlendingbot.readthedocs.io/en/latest/contributing.html)**
- [N/A] **I have updated the documentation in /docs if I have changed the config, arguments, logic in how the bot works, or anything that understandably needs a documentation change.**
- [N/A] **I have updated the config file accordingly if my change requires a new configuration setting or changes an existing one.**
- [ ] **I have tested the bot with no issues for 24 continuous hours. If issues were experienced, they have been patched and tested again.**

### CODE REVIEW

**Comment by laxdog** [comment]
Need to fix this. Happens when non-numpy percentile is used. The test for it must be using the wrong data.

```
> python lendingbot.py
Starting WebServer at 127.0.0.1 on port 33221 with template www
Started WebServer, lendingbot status available at http://127.0.0.1:33221/lendingbot.html
Welcome to Lending Bot on POLONIEX
Long  higher: WARN: Exception found when analysing markets, if this happens for more than a couple minutes please create a Github issue so we can fix it. Otherwise, you can ignore it. Error: Unknown format code 'f' for object of type 'str'
DEBUG: Class:<type 'exceptions.ValueError'> Args:("Unknown format code 'f' for object of type 'str'",)
DEBUG: Type:<type 'exceptions.ValueError'> Value:Unknown format code 'f' for object of type 'str' LineNo:294
Traceback (most recent call last):
  File "/home/dog/MACD_testing/modules/MarketAnalysis.py", line 294, in get_rate_suggestion
    self.get_percentile(rates, self.lending_style),
  File "/home/dog/MACD_testing/modules/MarketAnalysis.py", line 342, in get_percentile
    result = truncate(result, 6)
  File "/home/dog/MACD_testing/modules/Data.py", line 141, in truncate
    return float('{0:.{1}f}'.format(f, n))
ValueError: Unknown format code 'f' for object of type 'str'
```

**Comment by laxdog** [comment]
Just need a quick review to merge, @rnevet / @Evanito ?

File: tests/test_MarketAnalysis.py [hunk 1]
```diff
@@ -15,92 +18,102 @@
...
 from modules.MarketAnalysis import MarketAnalysis
-from modules.Configuration import FULL_LIST
+from modules.Configuration import get_all_currencies
 from modules.Poloniex import Poloniex
 import modules.Configuration as Config
 import modules.Data as Data
 
 Config.init('default.cfg', Data)
-api = Poloniex(Config.get("API", "apikey", None), Config.get("API", "secret", None))
+api = Poloniex(Config, None)
```
Comment (rnevet) [inline tests/test_MarketAnalysis.py]: No Logger?

Comment (laxdog) [reply tests/test_MarketAnalysis.py]: I don't think it's needed for testing.

**Pull request merged**